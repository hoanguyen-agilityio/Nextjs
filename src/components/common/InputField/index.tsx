// Third party
import { ComponentProps } from 'react';

// Components
import { InputCustom } from '@/components';

interface IInputField extends ComponentProps<typeof InputCustom> {
  label: string;
  errorMsg?: string;
  message?: string;
  classLabel?: string;
  classWrapper?: string;
  fieldId: string;
}

const InputField = ({
  label,
  errorMsg,
  message,
  classLabel,
  classWrapper,
  fieldId,
  ...rest
}: IInputField) => {
  const getMessage = () => {
    if (errorMsg) {
      return errorMsg;
    }

    return message;
  };
  return (
    <div className={classWrapper}>
      <label
        className={`text-lg text-black-900 dark:text-white ${classLabel}`}
        htmlFor={fieldId}
      >
        {label}
      </label>
      <InputCustom {...rest} id={fieldId} />
      {(errorMsg || message) && (
        <p
          className={`text-xsm mt-4 font-normal ${errorMsg ? 'text-red-500' : 'text-ratio-200 dark:text-grayBlue-400'}`}
        >
          {getMessage()}
        </p>
      )}
    </div>
  );
};

export default InputField;
