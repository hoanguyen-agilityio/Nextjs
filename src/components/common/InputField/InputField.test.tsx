import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import InputField from '.';

describe('InputField component', () => {
  test('renders with default type', () => {
    const { container } = render(
      <InputField
        label="InputField Test"
        errorMsg="Error message"
        fieldId="1"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
