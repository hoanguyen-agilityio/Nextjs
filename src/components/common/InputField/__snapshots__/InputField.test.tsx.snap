// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component renders with default type 1`] = `
<div>
  <div>
    <label
      class="text-lg text-black-900 dark:text-white undefined"
    >
      InputField Test
    </label>
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end"
      data-has-value="true"
      data-slot="base"
    >
      <div
        class="flex flex-col w-xl h-14"
        data-slot="main-wrapper"
      >
        <div
          class="relative tap-highlight-transparent flex-row shadow-sm gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 flex items-center space-x-2 h-full placeholder:text-ratio-300 dark:bg-gray-700 dark: text-ratio-700 dark:placeholder:text-ratio-700 focus-within:ring-0 py-4 px-6 w-full rounded-lg"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <div
            class="w-full items-center h-full box-border flex"
            data-slot="inner-wrapper"
          >
            <input
              aria-label=" "
              class="w-full bg-transparent !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground placeholder:text-ratio-300 dark:placeholder:text-ratio-700 dark: text-ratio-700 flex-grow focus:outline-none focus:ring-0 text-base font-semibold"
              data-slot="input"
              id="react-aria-:r0:"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <p
      class="text-xsm mt-4 font-normal text-red-500"
    >
      Error message
    </p>
  </div>
</div>
`;
